src/ip/tools.cc

```diff
@@ -55,8 +56,21 @@ Ip::ProbeTransport()
     debugs(3, 2, "Missing RFC 3493 compliance - attempting split IPv4 and IPv6 stacks ...");
     EnableIpv6 |= IPV6_SPECIAL_SPLITSTACK;
 #endif
-    // TODO: attempt to use the socket to connect somewhere ?
-    //  needs to be safe to contact and with guaranteed working IPv6 at the other end.
+
+    // Test for IPv6 loopback/localhost address binding
+    Ip::Address ip;
+    ip.setLocalhost();
+    if (ip.isIPv6()) { // false on --disable-ipv6 builds
```

Comment:
AFAICT, this check will always succeed because EnableIpv6 is always true here, and ip.setLocalhost() sets an IPv6 address when EnableIpv6 is true. Am I missing something?

The `false on --disable-ipv6 builds` comment implies that `USE_IPV6` has some effect on setLocalhos() and/or  isIPv6(), but I do not see that effect. Is it hidden somewhere deeper than the code of those two methods?

Reply:
Just me being extra paranoid. Since admin removing ::1 allocation is an edge case already I do not want this test to suddenly start disabling IPv6 if we or someones patch does anything in future that changes that situation (like looking up a local definition of localhost or patching setLocalhost() to an IPv4 address).
 With this the test will skip unless it is working with at least a valid IPv6 address. Even if it is not strictly ::1.

Reply:
If I interpret your answer correctly, then I was not missing any hidden dependencies, and the source code comment is indeed lying -- isIPv6() will _not_ return `false on --disable-ipv6 builds`. Please replace that comment. I suggest:

```suggestion
    if (ip.isIPv6()) { // paranoid; always succeeds if we got this far
```

Reply:
@yadij, please do not resolve change requests made by others.